MapboxMobileEvents/MMEConfigurationUpdater.m

```diff
@@ -0,0 +1,59 @@
+#import "MMEConfigurationUpdater.h"
+#import "MMEAPIClient.h"
+
+@interface MMEConfigurationUpdater ()
+
+@property (nonatomic) NSDate *configurationRotationDate;
+@property (nonatomic) MMEEventsConfiguration *configuration;
+
+@end
+
+@implementation MMEConfigurationUpdater
+
+- (instancetype)init {
+    NSAssert(false, @"Use `-[MMEConfigurationUpdater initWithTimeInterval:]` to create instances of this class.");
+    return nil;
+}
+
+- (instancetype)initWithTimeInterval:(NSTimeInterval)timeInterval {
+    if (self = [super init]) {
+        _timeInterval = timeInterval;
+    }
+    return self;
+}
+
+- (void)updateConfigurationFromAPIClient:(MMEAPIClient *)apiClient {
+    if (self.configurationRotationDate && [[NSDate date] timeIntervalSinceDate:self.configurationRotationDate] >= 0) {
+        self.configuration = nil;
+    }
+    if (!self.configuration) {
+        [apiClient getConfigurationWithCompletionHandler:^(NSError * _Nullable error, NSData * _Nullable data) {
+            if (!error) {
+                self.configuration = [MMEEventsConfiguration configuration];
+                [self parseJSONFromData:data];
```

Comment:
Could we combine these two calls into one?  e.g.
```
+[MMEEventsConfiguration configurationFromData:]
```

It's confusing as it is, and `parseJSONFromData` only modifies the configuration.